pkg/grpc/proxy/handler.go

```diff
@@ -230,23 +248,31 @@ func (s *handler) forwardClientToServer(src grpc.ClientStream, dst grpc.ServerSt
 func (s *handler) forwardServerToClient(src grpc.ServerStream, dst grpc.ClientStream, requestID string) chan error {
 	ret := make(chan error, 1)
 	go func() {
-		f := &codec.Frame{}
```

Comment:
Reviewers: this had to be moved within loop otherwise the replay buffer would capture stale data. Example:

```text
== APP == 2023/01/12 09:37:52 Received ping from client: 🦥 1
== APP == 2023/01/12 09:37:53 Received ping from client: 🦥 2
== APP == 2023/01/12 09:37:54 Received ping from client: 🦥 3
== APP == 2023/01/12 09:37:55 Received ping from client: 🦥 4
== APP == 2023/01/12 09:37:56 Received ping from client: 🦥 5
== APP == 2023/01/12 09:37:57 Received ping from client: 🦥 6
== APP == 2023/01/12 09:37:58 Received ping from client: 🦥 7
== APP == 2023/01/12 09:37:59 Received ping from client: 🦥 8
== APP == 2023/01/12 09:38:00 Received ping from client: 🦥 9
== APP == 2023/01/12 09:38:01 Received ping from client: 🦥 10
== APP == 2023/01/12 09:38:02 Received ping from client: 🦥 11
== APP == 2023/01/12 09:38:03 Received ping from client: 🦥 12
== APP == 2023/01/12 09:38:04 Received ping from client: 🦥 13
## ... (Restart server app)
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 14
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 15
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 16
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 17
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 18
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 19
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 20
== APP == 2023/01/12 09:38:24 Received ping from client: 🦥 21
```

As you can see, message "14" is repeated many times due to storing a pointer in the replay buffer

(Also this is an example of how, after a reconnection, the entire buffer is replayed back to the client every time and the target receives 14 messages that were already delivered)